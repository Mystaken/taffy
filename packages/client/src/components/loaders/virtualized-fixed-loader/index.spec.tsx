import React from 'react';
import { create } from 'react-test-renderer';
import { VirtualizedFixedLoader } from '.';
import { mount } from 'enzyme';

const Row = ({ index, style }: any) => {
  return (
    <div className={index % 2 ? 'ListItemOdd' : 'ListItemEven'} style={style}>
      Row {index}
    </div>
  );
};

describe('<VirtualizedFixedLoader />', () => {
  it('has rendered correctly', () => {
    const component = create(
      <div style={{ height: '500px' }}>
        <VirtualizedFixedLoader itemSize={35} itemCount={1000}>
          {Row}
        </VirtualizedFixedLoader>
      </div>
    );
    const tree = component.toJSON()!;
    expect(tree).toMatchSnapshot();
  });

  it('should moount', () => {
    mount(
      <div style={{ height: '500px' }}>
        <VirtualizedFixedLoader itemSize={35} itemCount={1000}>
          {Row}
        </VirtualizedFixedLoader>
      </div>
    );
  });
});
