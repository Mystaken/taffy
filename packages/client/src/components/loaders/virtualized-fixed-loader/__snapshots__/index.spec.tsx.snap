// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VirtualizedFixedLoader /> has rendered correctly 1`] = `
<div
  style={
    Object {
      "height": "500px",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "overflow": "auto",
        "width": "100%",
      }
    }
  >
    <style
      global={true}
      jsx={true}
    >
      
      ::-webkit-scrollbar {
        width: 0px; /* Remove scrollbar space */
        display: none;
      }
    
    </style>
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    />
  </div>
</div>
`;
